core/app/finders/spree/taxonomies/find.rb

```diff
@@ -0,0 +1,57 @@
+module Spree
+  module Taxonomies
+    class Find
+      def initialize(params = {})
+        @ability  = params[:ability]
+        @action   = params[:action]
+        @name     = params[:name]
+        @order    = params[:order]
+      end
+
+      def call
+        collection = by_ability(scope)
+        collection = by_name(collection)
```

Comment:
`collection = by_name(collection) unless name.nil?`
What do you think about moving checks into call method? They'll be pretty one liners and it will simplify `by_something` methods.

Reply:
Yes, we could do that, both approaches are fine for me. I just think that current implementation describes the flow of things much better - everybody can see which filters are applied and in which order and there are no conditionals therefore the flow is straightforward. Each separate method has a guard clause as a fallback which kinda makes sense to me + it's easy to extend each guard clause with additional logic. Let me know what you think